hw04/BarbieWalletBalance.cs

```diff
@@ -16,12 +16,48 @@ public class BarbieBank : MonoBehaviour
 
     void Start()
     {
-        int[] biggerWallet = BarbieWallet(500, 600);
+
+        //Question 2- Instantiate a new career for Barbie 
+        BarbieWorld<string> barbieWorld = new BarbieWorld<string>();
+        barbieWorld.BarbieCareers("GameDesigner");
+
+    //Problem 3
+        // Convert the existing function to a generic array 
+        var biggerWallet = BarbieWallet(500, 600);
```

Comment:
This problem is very close. Needs an open and close. When we call this function we want to pass in a function. We want to pass in some type - here it is an `int`.

```
    void Start()
    {
        int[] biggerWallet = BarbieWallet<int>(500, 600);
        Debug.Log(biggerWallet.Length + " " + biggerWallet[0] + " " + biggerWallet[1]);

        GetMoney<int>(500, 600);
    }

    // Generic function for BarbieWallet
    private T[] BarbieWallet<T>(T numOfPennies, T cashAmount)
    {
        return new T[] { numOfPennies, cashAmount };
    }

    // BarbieBank generic function
    private T[] BarbieBank<T>(T numOfPennies, T cashAmount, T numOfCreditCards)
    {
        return new T[] { numOfPennies, cashAmount, numOfCreditCards };
    }
```